### ISSUE TITLE
Track parameter estimation gives wrong momentum

### ISSUE THREAD

**Issue by beomki-yeo** [acts-project/traccc#427]
It's an extension of #413; I tried to reconstruct the momentum of the given spacepoint files provided by @shimasnd, which failed.

https://github.com/acts-project/traccc/blob/main/tests/cpu/test_seeding.cpp
```
    //...
    
    // Make sure that we have reasonable estimation on momentum
    /* Currently disabled
    EXPECT_NEAR(bound_params[0].p(), 16.62 * unit<scalar>::GeV,
                0.1 * unit<scalar>::GeV);
    */
    
    //...
    
    // Make sure that we have reasonable estimation on momentum
    /* Currently disabled
    EXPECT_NEAR(bound_params[0].p(), 1.85 * unit<scalar>::GeV,
                0.1 * unit<scalar>::GeV);
    */
    
    //...
```

Thing is that I get the right value when I did the same with spacepoints created by `detray::helix` :thinking:
https://github.com/acts-project/traccc/blob/main/tests/cpu/test_track_params_estimation.cpp

**Comment by beomki-yeo** [comment]
As a note, each test gives the momentum of 24.2 GeV (truth: 16.62 GeV) and 1.63 GeV (truth: 1.85 GeV)

**Comment by shimasnd** [comment]
Sorry for long silence. 
When I tried with traccc codes from Jun 20, I got one seed for the first test giving proper momentum (16.72 GeV) and 4 seeds for the second test giving (1.63 GeV, 1.66 GeV, 1.58 GeV, 1.76 GeV). I'm confused with that the first test giving different value from the reported value, while the second test looks consistent with what Beomki reports...

**Comment by beomki-yeo** [comment]
Hmm. There has not been a change in track parameter estimation algorithm since June 19 (#418). So we are supposed to have the same result. It is also weird that the traccc test gives only one seed for the second event while yours gives four :thinking:

**Comment by shimasnd** [comment]
I forgot that I changed seeding configuration for my setup. Our difference should comes from there. 

Now I tried to skip our famous multiple scattering condition (dCotThetaMinusError2) in the triplet_finding_helper.hpp not to reject any triplet. As mathematically expected from 5 spacepoints, I got 10 seeds for each event. Their pT values are:
first event: 24.1722, 17.718, 17.392, 15.4698, 14.5699, 13.8803, 12.3048, 19.28, 16.7183, 16.4234
second event: 1.66069, 1.65246, 1.59824, 1.58493, 1.76318, 1.85601, 1.72328, 1.64533, 1.62816, 1.6118
all should be in GeV. 

Now I guess that track parameter estimation doesn't have any issue and it is a matter of which triplet survives the conditions in triplet_finding_helper.hpp.
